TACC/Core-CMS
# FP-1415: Callout Pattern with Fixes sans Plugin

### To Do

<details>

- [x] Schedule review with Design.
- [x] Record Design feedback. _— in author's local notes_
- [x] Evaluate whether dev review is blocked by Design feedback. _— it's not_
- [x] Create ticket(s) for changes from Design. → [FP-1464](https://jira.tacc.utexas.edu/browse/FP-1464)
- [x] Create ticket to retire Callout plugin. → [FP-1465](https://jira.tacc.utexas.edu/browse/FP-1465)

</details>

### Overview

Support Callouts sans custom plugin: use existing plugins & markup-based CSS (#425).*

Also, fix clickable area (#423).

_\* This will let us retire and uninstall the callout plugin._

### Issue

[FP-1415](https://jira.tacc.utexas.edu/browse/FP-1415) (and sub-tasks)

### Demos

| Demo | Notes |
| - | - |
| [**Manual Pattern Library**][cep-ui] | supports `.c-callout` via snippet |
| [**Automatic Pattern Library**][kss-ui] | test of [ugly KSS software](https://github.com/tacc-wbomar/Core-CMS-Pattern-Library-KSS) |

### Screenshots

#### Of Demos

Styles for all expected markup and image combinations.

| Manual Pattern Library | Automatic Pattern Library |
| - | - |
| [component variations](https://user-images.githubusercontent.com/62723358/147126769-38ce21fa-98f6-46b0-87b1-6a38f67c9e88.jpeg) <br /> [DjangoCMS structure](https://user-images.githubusercontent.com/62723358/147126767-a95bfabd-90f5-4d35-999e-a6fd8d1b868b.png) | [component variations](https://user-images.githubusercontent.com/62723358/147127339-d0d66fa4-ceea-4e73-91cd-7b754dd30ae3.jpeg) |

#### Use Cases

Where images are pre-cropped before upload. (To test non-cropped images, see "Demos".)

| [Frontera Use Case][page-ui-1]<br />(wide) | [Frontera Use Case][page-ui-1]<br />(narrow) | [BrainMap Use Case][page-ui-2]<br />(wide) | [BrainMap Use Case][page-ui-2]<br />(narrow) |
| - | - | - | - |
| ![FP-1417 Frontera System Wide](https://user-images.githubusercontent.com/62723358/147126890-6ee94aef-8abd-43dc-b0b9-8f411ddf0e7b.png) | ![FP-1417 Frontera System Narrow](https://user-images.githubusercontent.com/62723358/147126887-a7bb3d72-a933-48f9-a503-12299b634984.png) | ![FP-1417 BrainMap Sleuth Wide](https://user-images.githubusercontent.com/62723358/147126886-82d2e4a3-19cf-44bd-9234-d5cee7e3ebd0.png) | ![FP-1417 BrainMap Sleuth Narrow](https://user-images.githubusercontent.com/62723358/147126882-99ae92bb-0ba1-46a4-b2ad-a629db7abcf6.png) |

#### Link Interaction

A callout is not required to be a link, but when it is, hover and activation UI should exist.

| Default State | Hover State | Active State |
| - | - | - |
| ![FP-1417 State Default](https://user-images.githubusercontent.com/62723358/147127924-21706d9d-0d4c-4219-ba38-0feca357f179.png) | ![FP-1417 State Hover](https://user-images.githubusercontent.com/62723358/147127926-a30a83e9-0e3a-414f-8493-d4c4d9003c89.png) | ![FP-1417 State Active](https://user-images.githubusercontent.com/62723358/147127923-d337ea98-d459-4cb8-b7e8-11ca8e632294.png) |

### Testing

0. Use remote server ([CEP][cep-ui], [Frontera][page-ui-1], [Brainmap][page-ui-2]) or [Test CMS Changes](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes) locally.
1. Review/Create plugin combo's with class `c-callout` on the outermost elements.
2. Check that UI[^1]—on wide screen—matches [wide Design UI (scroll down)][des-ui--wide] ([screenshot][des-ui-snap--wide])
3. Check that UI[^1]—on narrow screen—matches [narrow Design UI (scroll down)][des-ui--thin] ([screenshot][des-ui-snap--thin]).

[^1]: Any structures using an "Image" with a "Caption" may be skipped during testing. Design says we don't need to support caption for "Callout" (but it will be supported by a future pattern used in Newsletter and DesignSafe).

[cep-ui]: https://cep.tacc.utexas.edu/design-system/ui-patterns/c-callout/
[kss-ui]: http://tacc-wbomar-css-pattern-library-kss.surge.sh/section-components.html#kss-fullscreen-kssref-components-callout
[des-ui--wide]: https://xd.adobe.com/view/f3dad8d1-88ee-40eb-b8da-667b00e4f876-a689/screen/ff0348ff-577d-450a-bef0-5326f3f7c705/specs/
[des-ui--thin]: https://xd.adobe.com/view/f3dad8d1-88ee-40eb-b8da-667b00e4f876-a689/screen/cce23001-6fdd-47fe-bf6b-9e8161f55c85/specs/
[des-ui-snap--wide]: https://user-images.githubusercontent.com/62723358/147128294-13c393a5-c6cc-4520-97bd-da7eeff13956.png
[des-ui-snap--thin]: https://user-images.githubusercontent.com/62723358/147128284-e120fea3-8b89-45ae-ad92-e00e280c8822.png
[page-ui-1]: https://dev.fronteraweb.tacc.utexas.edu/system/
[page-ui-2]: https://pprd.brainmap.tacc.utexas.edu/tools/sleuth/
